Assets/Atlas games/Scripts/DoubleXPItem/DoubleXPManager.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+public class DoubleXPManager : MonoBehaviour
+{
+    static TimeAndDateResponseModel date;
+    static string extractedDate;
+    static string extractedTime;
+    static string dateTimeString;
+
+    public enum DoubleXpDuration
+    {
+        Day,
+        Hour
+    }
+
+    TimeAndDateResponseModel _globalDate;
+    static TimeSpan globalTimeDifference;
+
+    async void Awake()
+    {
+        DontDestroyOnLoad(gameObject);
+
+        await SetDate();
+    }
+
+    private static float newDifferenceSeconds = 0;
+    private DateTime _currentGlobalDateTime;
+
+    async Task SetDate()
+    {
+        _globalDate = await APIManager.instance.GetCurrentDateAndTime();
+        dateTimeString = _globalDate.datetime;
+        string time = dateTimeString.Substring(11, 8);
+        string dateString = dateTimeString.Substring(0, 10);
+        string finalDateTime = dateString + time;
+        _currentGlobalDateTime = ConvertedStringToDate(finalDateTime);
+        if (_currentGlobalDateTime > ConvertedStringToDate(GlobalValue.DoubleXpActivationTime)
+                .AddHours(GlobalValue.DoubleXPDuration))
+        {
+            GlobalValue.DoubleXp = false;
+        }
+        else
+        {
+            GlobalValue.DoubleXp = true;
+        }
+
+        if (GlobalValue.MainTimeDifference == 0)
+        {
+            GlobalValue.MainTimeDifference = (float)(_currentGlobalDateTime - DateTime.Now).TotalSeconds;
+        }
+
+        TimeSpan newDifference = _currentGlobalDateTime -
+                                 DateTime.Now.AddSeconds(GlobalValue.MainTimeDifference);
+        if (Math.Abs(newDifference.TotalSeconds) > 100)
+        {
+            newDifferenceSeconds = (float)newDifference.TotalSeconds;
+        }
+    }
+
+    static DateTime ConvertedStringToDate(string insertedTime)
+    {
+        try
+        {
+            int year = int.Parse(insertedTime.Substring(0, 4));
+            int month = int.Parse(insertedTime.Substring(5, 2));
+            int day = int.Parse(insertedTime.Substring(8, 2));
+            int hour = int.Parse(insertedTime.Substring(10, 2));
+            int minute = int.Parse(insertedTime.Substring(13, 2));
+            int second = int.Parse(insertedTime.Substring(16, 2));
+            return new DateTime(year, month, day, hour, minute, second);
+       // string format = "yyyy-MM-ddHH:mm:ss";
+       // return DateTime.ParseExact(insertedTime, format, null);
+        }
+        catch (Exception e)
+        {
+            Debug.Log(e);
+                         return new DateTime();
+        }
```

Comment:
میدونم سر این خیلی برو بیا داشتیم ولی منظورم اینطوری بود:
و اینکه دقت کردم فرمتی که انتخاب کردی ایننه : `yyyy-MM-ddHH:mm:ss` این اصلا اشکالی نداره ولی استاندارد نیست بین تاریخ و زمان یه T باید باشه، توی کد زیر مثالش هست اینو توی GlobalValue هم بزار لطفا
```suggestion
        DateTime parsedDate;
        DateTime.TryParseExact(insertedTime,"yyyy-MM-ddTHH:mm:ss",null,DateTimeStyles.None,out parsedDate);
        return parsedDate;
```